### PR TITLE
Fix: #21. Reorganize TLS parts.

### PR DESCRIPTION
**Pull request by ioggstream**
## This PR

Reorganize TLS parts:

- move all references to TLS usage to a separate section
- mention "https" scheme

### CODE REVIEW

**Comment by ioggstream** [comment]
@aaronpk I tried to reorganize the TLS parts in a single section. As a first-time reader, I found the continuous reminders to TLS confusing and inherently suggesting that TLS was only required for some flows.

**Comment by aaronpk** [comment]
Thanks, this is great, I'll discuss this with the other editors next time we talk!

**Comment by ioggstream** [comment]
Thanks @aaronpk As this is a draft PR I'm really open to improvements.

**Comment by ioggstream** [comment]
cc: @peppelinux

File: draft-ietf-oauth-v2-1.md [hunk 1]
```diff
@@ -2227,26 +2213,12 @@ audience), typically a single resource server (or a list of resource
...
-encryption MUST be applied in addition to the usage of TLS
-protection.  As a further defense against token disclosure, the
-client MUST validate the TLS certificate chain when making requests
-to protected resources, including checking the Certificate Revocation
-List (CRL) {{RFC5280}}.
+Risks related to token disclosure are mitigated by the mandatory
+usage of mechanisms providing
+confidentiality, authentication and integrity such as {{RFC8446}}
+(see {{tls-version}}). 
+if the client should be prevented from observing the contents of the token,
```
Comment (ioggstream) [inline draft-ietf-oauth-v2-1.md:2220]:
```suggestion
If the client should be prevented from observing the contents of the token,
```

File: draft-ietf-oauth-v2-1.md [hunk 2]
```diff
@@ -2227,26 +2213,12 @@ audience), typically a single resource server (or a list of resource
...
+token MUST be encrypted.
```
Comment (ioggstream) [inline draft-ietf-oauth-v2-1.md:2221]: @aaronpk not sure that an use-case requirement should use normative language (MUST)...

**Comment by ioggstream** [comment]
@aaronpk the IETF Using TLS in Applications wg is working on that  https://tools.ietf.org/html/draft-ietf-uta-rfc7525bis-00 
we could ping them too.  In the meanwhile I think this PR improves the readability of the current document.

**Comment by aaronpk** [comment]
I like the editorial improvements. Since this contains normative changes in #40 it is pending that discussion.

**Comment by aaronpk** [comment]
@ioggstream the group agreed to the normative change of requiring HTTPS for redirect URIs so I would like to merge this PR. Could you rebase it to fix the merge conflicts? Thanks!

**Comment by ioggstream** [comment]
@aaronpk sure, but it will take some days :P hope it's not an issue...

**Comment by aaronpk** [comment]
Closing in favor of #98

**Pull request closed without merging**